src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -2301,6 +2309,14 @@ private static bool IsQualifiedNameInUsingDirective(SemanticModel model, NameSyn
             return false;
         }
 
+        private static bool IsGlobalAliasQualifiedName(NameSyntax name)
+        {
+            // Checks whether the `global::` alias is applied to the name
```

Comment:
won't this let this happen?

namespace A
{
    using A.B;
    
    class D { C c; }

    namespace B
   {
       public class C {}
   }
}

before "using A.B;" won't be simplified. but now it might get simplified to "B" if there is global::A.B ?

Reply:
might or might not desirable behavior. but certainly, behavior changes if it actually happens.

Reply:
Note: this would be undesirable.  We used to do this and specifically added the block in place to not simplify the right side of usings because of user feedback that they prefer fully qualified names here.

I think it's ok to strip `"global::"`.  But i would block it at that.  We shouldn't do full simplification.

Reply:
This only allows `AliasQualifiedNameSyntax` nodes with the `global::` alias to simplify. `QualifiedNameSyntax` nodes, like `A.B`, are still blocked from simplification.

Reply:
Yes.  I understand that.  However, say there is `using X = global::System.Int32`.  This should only be simplified to `using X = System.Int32`.  it should *not* be simplified to `using X = Int32` (if you're in a namespace called 'System', nor should it be simplified to `using X = int` (which is illegal).

My point is: you're making it so that simplification will run here.  But we still need to make sure that simplification simplifies *properly*.